src/Runtime/include/OnnxMlir.h

```diff
@@ -0,0 +1,289 @@
+//===-------- RtMemRef.h - external RtMemRef C/C++ API call header --------===//
+//
+// Copyright 2019-2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains declaration of external RtMemRef data structures and
+// helper functions.
+//
+//===----------------------------------------------------------------------===//
+#ifndef __RTMEMREF_H__
+#define __RTMEMREF_H__
+
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdbool.h>
+#include <stdint.h>
+#endif
+
+/*! \mainpage ONNX-MLIR Model Execution API documentation
+ *
+ * \section intro_sec Introduction
+ *
+ * ONNX-MLIR project comes with an executable `onnx-mlir` capable
+ * of compiling onnx models to a shared library. In this documentation, we
+ * explain and document ways to interact programmatically with the compiled
+ * shared library using the API's defined in `src/Runtime/include/OnnxMlir.h`.
+ *
+ * \section Important Data Structures
+ * \subsection RtMemRef
+ *
+ * `RtMemRef` is the data structure used to describe the runtime information
+ * (rank, shape, data type, etc) associated with a tensor (or `MemRef`, in the
+ * context of MLIR). Specifically, it can be described as a struct with the
+ * following data members:
+ *
+ * ```cpp
+ * void *_data;            // data buffer
+ * void *_alignedData;     // aligned data buffer that the rmr indexes.
+ * INDEX_TYPE _offset;     // offset of 1st element
+ * INDEX_TYPE *_dataSizes; // sizes array
+ * int64_t *_dataStrides;  // strides array
+ * int _dataType;          // ONNX data type
+ * int _rank;              // rank
+ * std::string _name;      // optional name for named access
```

Comment:
Using char * for getter/setter is because the API needs to work for both C and C++. Using std::string for _name internally is because the RtMemRef destructor will automatically call string destructor to free _name (done by the compiler) when the RtMemRef struct is destroyed. If char * is used for _name, RtMemRef destructor needs to free _name explicitly to avoid memory leak.

Reply:
It feels like adding name-based query is too complicated for C runtime API (the implementation as well as specification of semantics are both way more complicated than we should be worrying about at this moment, I think, due to the lack of dictionary-like standard container in c), maybe I'll make that happen for C++ API next.